features/request_specs/request_spec.feature

```diff
@@ -152,3 +152,23 @@ Feature: request spec
       """
     When I run `rspec spec`
     Then the example should pass
+
+  Scenario: testing subdomain constrained requests
+    Given a file named "spec/requests/widgets_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe "Widget management", :type => :request do
+        before { host! "api.example.com" }
+
+        it "creates a Widget" do
+          headers = { "ACCEPT" => "application/json" }
+          post "/widgets", :params => { :widget => {:name => "My Widget"} }, :headers => headers
+
+          expect(response.content_type).to eq("application/json; charset=utf-8")
```

Comment:
As you may have noticed, `content_type` depends on the Rails version. We have different examples, and they are either run or skipped depending on CI job matrix.
I suggest `.to start_with("application/json")` so that this example passes with all Rails versions from our matrix.

Reply:
Hi @pirj ,

Thanks for your feedback. I added the requested changes. 
However the test without the `host!` is still passing, so I wanna look into this again. I will let you know once the PR is ready for review again.
Thanks for taking the time!

Reply:
I checked on Rails 6 that this fails without `host!`